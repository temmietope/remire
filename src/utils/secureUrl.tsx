/**
 * @description takes url replaces http with https.
 */
export default (url: string): string => url.replace(/^(https?)/, 'https');
