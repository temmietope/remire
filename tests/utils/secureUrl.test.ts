// @ts-nocheck
import secureUrl from '../../src/utils/secureUrl';

const list = ['https://foo.bar', 'http://foo.bar', 'https://foohttp.bar'];

describe('convert http to https', () => {
  for (const [k, url] of Object.entries(list)) {
    it(`get secure url for ${k}`, () => {
      expect(secureUrl(url)).toMatchSnapshot();
    });
  }
});
