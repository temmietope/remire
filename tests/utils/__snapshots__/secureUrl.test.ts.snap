// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert http to https get secure url for 0 1`] = `"https://foo.bar"`;

exports[`convert http to https get secure url for 1 1`] = `"https://foo.bar"`;

exports[`convert http to https get secure url for 2 1`] = `"https://foohttp.bar"`;
